subsys/net/ip/net_context.c

```diff
@@ -924,6 +1011,10 @@ static int sendto(struct net_pkt *pkt,
 		return -EINVAL;
 	}
 
+	context->send_cb = cb;
```

Comment:
As far as I can tell (it's hard to tell, because there're many mixed-up commits here), this duplicates the code which already exists at https://github.com/zephyrproject-rtos/zephyr/pull/7118/files#diff-0fca2d20b5935f1677a31c9a6e2b5087L969 . I may imagine the idea was to move the code chunk up.